docs/_docs/03-ug-mutable-dataset.md

```diff
@@ -99,4 +107,127 @@ hs.refreshIndex("empIndex", "incremental")
 ```
 
 ### Hybrid Scan
-TODO
+
+Hybrid Scan utilizes existing index data along with newly appended source files and/or deleted
+source files, without explicit refresh operation. For an index with appended source files,
+HybridScan changes the query plan to shuffle new data on-the-fly and merge it with index records.
+For an index with deleted source files, Hyperspace also modifies the plan to exclude the rows from
+deleted files in the index data. This requires enabling lineage for the index at its creation time.
+
+Currently, HybridScan is disabled by default. You can check the [configuration](https://microsoft.github.io/hyperspace/docs/ug-configuration/)
+page to see how it can be enabled.
+
+If a dataset has many deleted source files, query performance could degrade when Hybrid Scan is
+enforcing deletes at the query runtime. Hyperspace provides two different configurations to tune
+this behavior. 
+You can use them to enable supporting deletes during Hybrid Scan (spark.hyperspace.index.hybridscan.delete.maxNumDeleted)
+and determine when it should be applied, depending on the total number of deleted source files (spark.hyperspace.index.hybridscan.delete.enabled).
+You can check the detail in [Append and Delete](#append-and-delete-dataset) section.
```

Comment:
nit: can we use "\`" for configs? `spark.hyperspace.index.hybridscan.delete.maxNumDeleted` to make it stand out?